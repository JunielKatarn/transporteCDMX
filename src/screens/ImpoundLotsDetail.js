import React from 'react'
import { FlatList, View } from 'react-native'
import { keyExtractor } from '../utils/list'
import ImpoundLot from '../components/ImpoundLot'
import { sharedStyles } from '../styles'
import SimplePlaceholder from '../components/SimplePlaceholder'
import { translate } from '../i18n'

const _renderItem = ({ item }) => <ImpoundLot impoundLot={item} />
const _renderHeader = () => <View style={sharedStyles.mb3} />

const _renderEmptyPlaceholder = () => (
  <SimplePlaceholder
    imageURL={require('../assets/images/eco_car.png')}
    text={translate('no_impound_lot')}
    fullHeight
  />
)

const ImpoundLotsDetail = props => {
  const { impoundLots } = props.route.params
  return (
    <FlatList
      data={impoundLots.reverse()}
      renderItem={_renderItem}
      keyExtractor={keyExtractor}
      ListHeaderComponent={_renderHeader}
      ListEmptyComponent={_renderEmptyPlaceholder}
    />
  )
}

export default ImpoundLotsDetail
