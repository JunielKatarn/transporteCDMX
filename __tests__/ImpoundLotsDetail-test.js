import React from 'react'
import { shallow } from 'enzyme'
import ImpoundLotsDetail from '../src/screens/ImpoundLotsDetail'
import impoundLotsFixture from './fixtures/impoundLots-fixture'
import navigation from './fixtures/navigation-fixture'

const params = { params: { impoundLots: impoundLotsFixture } }
const emptyParams = { params: { impoundLots: [] } }

describe('ImpoundLotsDetail', () => {
  it('it renders correctly WITH Impound Lot details', () => {
    const wrapper = shallow(<ImpoundLotsDetail navigation={navigation} route={params} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('it renders correctly WITHOUT Impound Lot details', () => {
    const wrapper = shallow(<ImpoundLotsDetail navigation={navigation} route={emptyParams} />)
    expect(wrapper).toMatchSnapshot()
  })
})
