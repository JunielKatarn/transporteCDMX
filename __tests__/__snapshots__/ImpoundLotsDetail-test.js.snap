// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImpoundLotsDetail it renders correctly WITH Impound Lot details 1`] = `
<FlatList
  ListEmptyComponent={[Function]}
  ListHeaderComponent={[Function]}
  data={
    Array [
      Object {
        "address": "This is an address",
        "code": "2",
        "coordinates": Object {
          "latitude": 19,
          "latitudeDelta": 0.01,
          "longitude": -79,
          "longitudeDelta": 0.01,
        },
        "date": "2020-01-01",
        "formattedDate": "2020-01-01 00:00",
        "formattedStatus": "Status",
        "hour": "00:00",
        "id": 1234567,
        "name": "This a name",
        "phone": "5555555",
        "showMap": false,
      },
      Object {
        "address": "This is an address",
        "code": "0",
        "coordinates": Object {
          "latitude": 19,
          "latitudeDelta": 0.01,
          "longitude": -79,
          "longitudeDelta": 0.01,
        },
        "date": "2020-01-01",
        "formattedDate": "2020-01-01 00:00",
        "formattedStatus": "Status",
        "hour": "00:00",
        "id": 1234567,
        "name": "This a name",
        "phone": "5555555",
        "showMap": true,
      },
    ]
  }
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;

exports[`ImpoundLotsDetail it renders correctly WITHOUT Impound Lot details 1`] = `
<FlatList
  ListEmptyComponent={[Function]}
  ListHeaderComponent={[Function]}
  data={Array []}
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;
